Example-CustomUserGroups-REXX/USRGRP01.rexx

```diff
@@ -0,0 +1,2980 @@
+FALSE = 0
+
+CUSTOM_USR_GRPS_BASE_URI = '/api/groups'
+
+/**********************************************************************
+ *                       Global Variables
+ **********************************************************************/
+/* JSON parser specific, enabled via -v */
+VERBOSE = FALSE
+
+hwiHostRC = 0
+HWIHOST_ON = FALSE
+
+/* default to TSO/E, enable via -I */
+ISVREXX = FALSE
+PARSER_INIT = FALSE
+
+/*
+ * CPC globals
+ */
+CPCname = ""
+CPCuri = ""
+CPCtargetName = ""
+localCPC = TRUE
+
+/*
+ * LPAR globals
+ */
+LPARname = ""
+lparURI = ""
+lparTargetName = ""
+localLPAR = TRUE
+
+/*
+ * Misc. globals
+ */
+targetedUserGroup = ""
+isMissingTargetGroup = TRUE
+
+addLPARToCustomUserGroup = FALSE
+removeLPARFromCustomUserGroup = FALSE
+
+parserHandle = ""
+
+customUserGroups. = 0
+customUserGroupMems. = 0
+
+/**********************************************************************
+ *                       Initial Entry Point
+ **********************************************************************/
+Parse Arg arguments
+Return Main(arguments)
+
+/**********************************************************************
+ *                       Main Function begins
+ **********************************************************************/
+Main:
+  argString = Arg(1)
+
+  /********************************************************************
+   * Get the arguments specified by the user
+   ********************************************************************/
+  If GetArgs(argString) <> 0 Then
+    Exit -1
+
+  Say 'Starting LSTGRPS for CPC name:' || CPCname ||,
+      ' and Customer User Group name:' || targetUserGroup
+
+  If ISVREXX Then Do
+    Say 'Running in an ISV REXX environment'
+
+    hwiHostRC = hwihost("ON")
+    Say 'HWIHOST("ON") return code is :(' || hwiHostRC || ')'
+
+    If hwiHostRC <> 0 Then
+      Exit FatalErrorAndCleanup('** unable to turn on HWIHOST **')
+
+    HWIHOST_ON = TRUE
+  End
+
+  /********************************************************************
+   * Include the constants for the JSON parser and MACLIB BCPii
+   * constants
+   ********************************************************************/
+  Call IncludeConstants
+  Call GetJSONToolkitConstants
+
+  If RESULT <> 0 Then
+    Exit FatalErrorAndCleanup( '** Environment error **' )
+
+  PROC_GLOBALS = 'VERBOSE parserHandle ' || HWT_CONSTANTS
+
+  /********************************************************************
+   * Create a new parser instance
+   ********************************************************************/
+  Call InitJSONParser
+  If RESULT <> 0 Then
+    Exit FatalErrorAndCleanup( '** Parser init failure **' )
+
+  /********************************************************************
+   * Grab the CPC Information
+   ********************************************************************/
+  If localCPC Then Do
+    If GetLocalCPCInfo() <> TRUE Then
+      Exit FatalErrorAndCleanup( '** failed to get local CPC info **' )

```

Comment:
Double check the message capitalization consistency.  L248 is capitalized but this is not, not sure if this was intended.